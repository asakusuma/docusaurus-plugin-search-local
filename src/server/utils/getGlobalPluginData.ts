import { GlobalPluginData } from "docusaurus-plugin-search-local";
import { PluginConfig } from "../../types";

export function getGlobalPluginData(
  pluginConfig: PluginConfig
): GlobalPluginData {
  const {
    externalSearchSources,
    removeDefaultStopWordFilter,
    searchResultContextMaxLength,
    searchResultLimits,
    translations,
  } = pluginConfig;

  return {
    externalSearchSources,
    indexHash: null, // TODO: generate index hash
    removeDefaultStopWordFilter,
    searchResultContextMaxLength,
    searchResultLimits,
    translations,
  };
}
