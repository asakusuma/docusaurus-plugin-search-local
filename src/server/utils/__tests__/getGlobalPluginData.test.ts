import { PluginConfig } from "../../../types";
import { getGlobalPluginData } from "../getGlobalPluginData";

describe("getGlobalPluginData", () => {
  const baselinePluginConfig: PluginConfig = {
    id: "foo",
    indexDocs: true,
    indexBlog: true,
    indexPages: false,
    docsRouteBasePath: ["docs"],
    blogRouteBasePath: ["blog"],
    language: ["en"],
    hashed: false,
    docsDir: ["docs"],
    blogDir: ["blog"],
    removeDefaultStopWordFilter: false,
    highlightSearchTermsOnTargetPage: false,
    searchResultLimits: 8,
    searchResultContextMaxLength: 50,
    ignoreFiles: [],
    translations: {
      search_placeholder: "Search",
      see_all_results: "See all results",
      no_results: "No results.",
      search_results_for: 'Search results for "{{ keyword }}"',
      search_the_documentation: "Search the documentation",
      count_documents_found_plural: "{{ count }} documents found",
      count_documents_found: "{{ count }} document found",
      no_documents_were_found: "No documents were found",
    },
    externalSearchSources: [],
  };

  test("it should generate global plugin data", () => {
    const config: PluginConfig = { ...baselinePluginConfig };

    expect(getGlobalPluginData(config)).toEqual({
      externalSearchSources: config.externalSearchSources,
      indexHash: null,
      language: config.language,
      removeDefaultStopWordFilter: config.removeDefaultStopWordFilter,
      searchResultContextMaxLength: config.searchResultContextMaxLength,
      searchResultLimits: config.searchResultLimits,
      translations: config.translations,
    });
  });

  // TODO: write a test where an index hash is actually generated.
});
