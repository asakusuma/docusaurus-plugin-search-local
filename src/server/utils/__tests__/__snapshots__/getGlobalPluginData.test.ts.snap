// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGlobalPluginData it should generate an index hash when enabled 1`] = `
Object {
  "externalSearchSources": Array [],
  "indexHash": "aa634aa9",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": Object {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for \\"{{ keyword }}\\"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;

exports[`getGlobalPluginData it should only index blog when enabled 1`] = `
Object {
  "externalSearchSources": Array [],
  "indexHash": "05cf5c9c",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": Object {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for \\"{{ keyword }}\\"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;

exports[`getGlobalPluginData it should only index docs when enabled 1`] = `
Object {
  "externalSearchSources": Array [],
  "indexHash": "779c6e82",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": Object {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for \\"{{ keyword }}\\"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;
